In [2]:
import sys
import os
sys.path.append(os.path.abspath(os.path.join(os.getcwd(), "../")))

from database_operations import DatabaseOperations
from overtake_model import OvertakingModel
from race_data import RaceDataSetup
from race_dataframe import RaceDataframe
from race_sim import RaceSimulation
from evaluation import RaceSimEvaluation, EvaluateMany

In [3]:


evaluator = EvaluateMany()

# Add races to the evaluator
for year, circuit in [(2024, "Sakhir"), (2024, "Monza"), (2024, "Melbourne"), (2024, "Austin")]:
    db = DatabaseOperations(year, circuit)
    race_data = RaceDataframe(db)
    race_setup = RaceDataSetup(db, race_data)
    overtake = OvertakingModel(race_data.race_df)
    simulation = RaceSimulation(race_setup, overtake)
    simulation.simulate()

    evaluator.add_race(simulation, race_data, db)

# Evaluate all races
evaluator.evaluate_all_races()

aggregated_results = evaluator.get_aggregated_results()
(aggregated_results)


Evaluating race 1...
Evaluating race 2...
Evaluating race 3...
Evaluating race 4...


{'average_mae': np.float64(456.729647426279),
 'average_position_accuracy': 0.0912828947368421,
 'average_top_3_accuracy': 0.3333333333333333,
 'average_mean_error': 4.535690789473684,
 'average_total_error': 83.0,
 'average_simulated_gap_to_front': np.float64(2612.0041972280515),
 'average_actual_gap_to_front': np.float64(45259.052749999995)}

{'average_mae': np.float64(239.42474557592521),
 'average_position_accuracy': 0.12384868421052632,
 'average_top_3_accuracy': 0.5,
 'average_mean_error': 3.3347039473684212,
 'average_total_error': 61.5,
 'average_simulated_gap_to_front': np.float64(1241.6709643077284),
 'average_actual_gap_to_front': np.float64(45259.052749999995)}